packages/flutter/lib/src/material/carousel.dart

```diff
@@ -447,60 +448,68 @@ class _CarouselViewState extends State<CarouselView> {
   }
 
   Widget _buildSliverCarousel(ThemeData theme) {
-    if (_itemExtent != null) {
-      return _SliverFixedExtentCarousel(
-        itemExtent: _itemExtent!,
-        minExtent: widget.shrinkExtent,
-        delegate: SliverChildBuilderDelegate((BuildContext context, int index) {
-          return _buildCarouselItem(theme, index);
-        }, childCount: widget.children.length),
-      );
-    }
-
-    assert(
-      _flexWeights != null && _flexWeights!.every((int weight) => weight > 0),
-      'flexWeights is null or it contains non-positive integers',
+  if (_itemExtent != null) {
+    return _SliverFixedExtentCarousel(
+      itemExtent: _itemExtent!,
+      minExtent: widget.shrinkExtent,
+      delegate: SliverChildBuilderDelegate((BuildContext context, int index) {
+        if (widget.isInfinite) {
+          index %= widget.children.length;  // Infinite scrolling logic
+        }
+        return _buildCarouselItem(theme, index);
+      }, childCount: widget.isInfinite ? null : widget.children.length),
     );
-    return _SliverWeightedCarousel(
+  }
+
+  assert(
+    _flexWeights != null && _flexWeights!.every((int weight) => weight > 0),
+    'flexWeights is null or it contains non-positive integers',
+  );
+
+  return _SliverWeightedCarousel(
       consumeMaxWeight: _consumeMaxWeight,
       shrinkExtent: widget.shrinkExtent,
       weights: _flexWeights!,
       delegate: SliverChildBuilderDelegate((BuildContext context, int index) {
+        if (widget.isInfinite) {
+          index %= widget.children.length;  // Infinite scrolling logic
+        }
         return _buildCarouselItem(theme, index);
-      }, childCount: widget.children.length),
+      }, childCount: widget.isInfinite ? null : widget.children.length),
```

Comment:
This does not look right, even though they loop around, it is still the same number of children, correct?